First stab at merging uploaded data into master table

scog1234 opened the pull request:
Right now, this doesn't handle the case where there is no pre-existing master table. Also, there seems to be a bug in that IsSimulated is getting set even for uploads where the box wasn't checked.

scog1234 committed:
Append uploaded data to master table.

 At the moment, throws an exception if there is no master table.
 This needs fixing, once I figure out how.

scog1234 committed:
Merge with upstream

Merge branch 'master' of https://github.com/keflavich/flask_project

scog1234 committed:
Keep the plotting code together in one place

keflavich commented on upload_form.py:
```diff
@@ -225,9 +224,23 @@ def set_columns(filename, fileformat=None):
+    if os.path.isfile(merged_table_name):
+        merged_table = Table.read(merged_table_name, format='ascii.ipac')
+    else:
+        raise Exception("Can't find merged_table.ipac!")
```
use `IOError`

keflavich commented on upload_form.py:
```diff
@@ -225,9 +224,23 @@ def set_columns(filename, fileformat=None):
+    else:
+        raise Exception("Can't find merged_table.ipac!")
+#        merged_table = Table(data=None,
+#                       names=['Names','IDs','SurfaceDensity','VelocityDispersion','Radius','IsSimulated'])
```
`merged_table = Table(data=None, names=['Names','IDs','SurfaceDensity','VelocityDispersion','Radius','IsSimulated'], dtype=['str','str','float','float','float','bool'])`

keflavich commented:
@scog1234 ping me with an @keflavich when you're satisfied with the PR (especially the line noted)

scog1234 committed:
Create a new master table if one doesn't already exist.

 Currently, maximum length of username and ID are hard-coded to 64 characters.
 This needs to be adjusted to something more reasonable in the future. Also,
 we seem to be missing the units data still.

scog1234 committed:
Set units in newly created master table

scog1234 committed:
Actually reorder the columns

scog1234 committed:
Correctly record whether or not datapoint is from simulated data
